core/src/main/java/brooklyn/util/crypto/SecureKeys.java

```diff
@@ -19,182 +19,149 @@
-    public static TrustManager getTrustManager(KeyStore trustStore) {
-        return getTrustManager(trustStore, null);
-    }
-    /** returns the trust manager inferred from trustStore, matching the type (if not null);
-     * throws exception if there are none, or if there are multiple */
-    @SuppressWarnings("unchecked")
-    public static <T extends TrustManager> T getTrustManager(KeyStore trustStore, Class<T> type) {
-        try {
-            TrustManagerFactory tmf = TrustManagerFactory.getInstance(
-                    TrustManagerFactory.getDefaultAlgorithm());
-            tmf.init(trustStore);
-            T result = null;
-            for (TrustManager tm: tmf.getTrustManagers()) {
-                if (type==null || type.isInstance(tm)) {
-                    if (result!=null)
-                        throw new IllegalStateException("Multiple trust managers matching "+type+" inferred from "+trustStore);
-                    result = (T)tm;
-                }
-            }
-            if (result!=null)
                 return result;
-            throw new NoSuchElementException("No trust manager matching "+type+" can be inferred from "+trustStore);
-        } catch (Exception e) {
-            throw Exceptions.propagate(e);
-        }
-    }
-    
-    public static X509TrustManager getTrustManager(X509Certificate certificate) {
-        try {
-            KeyStore ks = newKeyStore();
-            ks.setCertificateEntry("", certificate);
-            return getTrustManager(ks, X509TrustManager.class);
-        } catch (KeyStoreException e) {
-            throw Exceptions.propagate(e);
+
+            } catch (Exception e2) {
+                Exceptions.propagateIfFatal(e2);
+                throw Exceptions.propagate(e);
+            }
         }
     }
 
-    /** converts a certificate to the canonical implementation, commonly sun.security.x509.X509CertImpl,
-     * which is required in some places -- the Bouncy Castle X509 impl is not accepted 
-     * (e.g. where certs are chained, passed to trust manager) */
-    public static X509Certificate getCanonicalImpl(X509Certificate inCert) {
-        try {
-            KeyStore store = SecureKeys.newKeyStore();
-            store.setCertificateEntry("to-canonical", inCert);
-            ByteArrayOutputStream out = new ByteArrayOutputStream();
-            store.store(out, "".toCharArray());
-
-            KeyStore store2 = KeyStore.getInstance(KeyStore.getDefaultType());
-            store2.load(new ByteArrayInputStream(out.toByteArray()), "".toCharArray());
-            return (X509Certificate) store2.getCertificate("to-canonical");
-        } catch (Exception e) {
-            throw Exceptions.propagate(e);
-        }
+    /** because KeyPair.equals is not implemented :( */
+    public static boolean equal(KeyPair k1, KeyPair k2) {
+        return Objects.equal(k2.getPrivate(), k1.getPrivate()) && Objects.equal(k2.getPrivate(), k1.getPrivate());
```

Comment:
Looks wrong. Did you mean for one of the `Objects.equal` to compare the `getPublic()`?